src/vtp/web/api/backend.py

```diff
@@ -0,0 +1,219 @@
+"""
+Backend support for the web-api.  One important aspect of this file is
+to support web-api testing in the absense of a live ElectionData
+deployment.  An ElectionData deployment is when a
+setup-vtp-demo-operation operation has been run, nominally creating a
+/opt/VoteTrackerPlus/demo.01 ElectionData folder and the required
+subfolders.
+
+With an ElectionData deployment the VTP git commands can be executed
+and VoteTracker+ can function as designed.
+
+Without an ElectionData deployment VTP git commands cannot be
+executed.  Currently this state is configured by the _MOCK_MODE
+variable below.  When set, and when this repo is part of the
+VTP-dev-env parent repo (or when the VoteTrackerPlus and
+VTP-mock-election.US.xx repos are simply sibling repos of this one),
+the commands here do not call into the VoteTrackerPlus repo and
+instead stub out the effective IO operations with static, non-varying
+mock data.  That mock data is currently nominally stored in (checked
+into) the VTP-mock-election.US.xx repo as the mock data is a direct
+function of the live ElectionData election configuration and CVR data
+found in that repo.  And that is due to two things: 1) the
+VTP-mock-election.US.xx holds the configuration of an election such as
+the blank ballot definition and 2) it also holds several hundred
+pre-cast random ballots so to fill the ballot cache so that
+ballot-checks can be immediately produced upon casting a ballot.
+
+Regardless, for the time being the VTP-mock-election.US.xx also holds
+checkedin mock values for the data that the web-api and above layers
+need when running in mock mode.
+"""
+
+import csv
+import json
+
+from vtp.core.common import Common
+from vtp.ops.accept_ballot_operation import AcceptBallotOperation
+from vtp.ops.cast_ballot_operation import CastBallotOperation
+from vtp.ops.setup_vtp_demo_operation import SetupVtpDemoOperation
+from vtp.ops.show_contests_operation import ShowContestsOperation
+from vtp.ops.tally_contests_operation import TallyContestsOperation
+from vtp.ops.verify_ballot_receipt_operation import VerifyBallotReceiptOperation
+
+
+class VtpBackend:
+    """Class to keep the namespace separate"""
```

Comment:
Actually, I initially created this file because I thought that Neil wanted this code removed from main.py (which is fine).  I created a class just to keep the namespace separate from main.py as I did not want to collide with that namespace - I did/do not know what else may be going in there.  So my reasons for a class are weak.

Reply:
done